### PR TITLE
Added start and end time to type process. Fixes #742

### LINKED ISSUE #742: Add process_start/end_time to process.json
**For which schema is a change/update being suggested?**

I would like to request an update to the `process.json` type schema.

**What should the change/update be?**

I would like to add the new fields - `process_start_time`, `process_end_time` - to this schema to allow data contributors to add when a process began/finished.

**What new field(s) need to be changed/added?**

1.
* Field name: process_start_time
* Field description: When the process started, in date-time format, yyyy-mm-ddThh:mm:ssZ.
* Required: no
* Example: 2019-01-16T12:00:00Z
* Guideline: Only include if providing the time doesn't make a donor re-identifiable.
2.
* Field name: process_end_time
* Field description: When the process finished, in date-time format, yyyy-mm-ddThh:mm:ssZ.
* Required: no
* Example: 2019-01-16T12:00:00Z
* Guideline: Only include if providing the time doesn't make a donor identifiable.

**Why is the change requested?**

A data contributor would like to include the start/end times for a process to help identify possible batch effects as their processes were completed months apart. A discussion needs to be had about whether or not to reintroduce these fields with the guideline to only include if they don't make a donor identifiable.

### PR DESCRIPTION
**Pull request by willrockout**
### Release notes
For ```process.json``` schema:

-  Added `start_time`

- Added `end_time`

### Reviews requested

- Need 2 Reviewers to approve because this is a minor update
- Last Reviewer to review/approve, merge changes by 26/04/19

### CODE REVIEW

File: json_schema/type/process/process.json [hunk 1]
```diff
@@ -40,6 +40,22 @@
             "$ref": "core/process/process_core.json",
             "user_friendly": "Process core"
         },
+        "start_time": {
```
Comment (malloryfreeberg) [inline json_schema/type/process/process.json]: Since this is a date-time field, should the field names be `start_date_time` and `end_date_time`?

Comment (willrockout) [reply json_schema/type/process/process.json]: Good point! I'm torn on this as the example driving this addition has just a difference in date with no time provided so I'm still on the fence for the mandatory date-time format.

Comment (malloryfreeberg) [reply json_schema/type/process/process.json]: This seems related to the issue https://github.com/HumanCellAtlas/metadata-schema/issues/539 where we need to be able to validate either partial dates, just dates, or date-times. I liked @daniwelter 's idea of having our own regex and implementing `oneOf`: https://github.com/HumanCellAtlas/metadata-schema/issues/539#issuecomment-465535372

Maybe we can do that here?

Comment (malloryfreeberg) [reply json_schema/type/process/process.json]: Updating with `oneOf` to allow either `date-time` or a regex that matches YYYY, YYYY-MM, or YYYY-MM-DD (as per https://github.com/HumanCellAtlas/metadata-schema/issues/966).

File: json_schema/update_log.csv [hunk 2]
```diff
@@ -1 +1,3 @@
 Schema,Change type,Change message,Version,Date
+type/process/process.json,minor,"Added start_date_time to process.json. Fixes #742.",,
```
Comment (malloryfreeberg) [inline json_schema/update_log.csv]: Remove "`.json`" from the Schema bit

**Comment by willrockout** [comment]
@malloryfreeberg @cjvillar Fixed update log can you review

File: json_schema/update_log.csv [hunk 3]
```diff
@@ -1 +1,3 @@
 Schema,Change type,Change message,Version,Date
+type/process/process,minor,"Added start_date_time to process.json. Fixes #742.",,
+type/process/process,minor,"Added end_date_time to process.json. Fixes #742.",,
```
Comment (cjvillar) [inline json_schema/update_log.csv]: We should be more conservative with our version increments. Technically these changes are related so we could squeeze them in one line. Also, we should try to keep the update log as accurate as possible, " added start_time"  But I approved and merged.

Comment (malloryfreeberg) [reply json_schema/update_log.csv]: I agree with @cjvillar , that adding these two related, optional fields should constitute a single minor version increment, not two. And - for consistency in the changelog - I would request that `start_date_time` and `end_date_time` be replaced with the actual names of the new fields.

Comment (daniwelter) [reply json_schema/update_log.csv]: Butting in here to say +1 on limiting version number increments. Separate lines in the update log should ideally only apply to the same schema is there are different change types, and even then, the highest change type should carry the others.

**Comment by malloryfreeberg** [comment]
@cjvillar you're up to review/merge!

**Pull request merged**